tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -4023,6 +4023,180 @@ TEST_F(InputProcessorFixture, clean_epjson)
     EXPECT_EQ(expected, cleanstring);
 }
 
+TEST_F(InputProcessorFixture, reportIDFRecordsStats_basic)
```

Comment:
Unit tests: 

* One with no extensible fields. Also tests a truncated object (has fields with all defaults at the end that just aren't provided in the IDF).
* One with extensible fields.

Tests are heavily commented to show how I counted objects to create the test (instead of just throwing the numeric values in the EXPECT_EQ)